tests/test_hierarchy.cc

```diff
@@ -203,7 +203,8 @@ BOOST_AUTO_TEST_CASE(benchmark)
 
 BOOST_AUTO_TEST_CASE(benchmark_mf)
 {
-  dealii::MultithreadInfo::set_thread_limit(1);
+  dealii::MultithreadInfo::set_thread_limit(
+      dealii::numbers::invalid_unsigned_int);
```

Comment:
I am getting confused on how we control multithreading.  Would you enlighten us on the precedence rule in deal.II, `DEAL_II_NUM_THREADS ` environment variable VS 3rd argument of `Utilities::MPI::MPI_InitFinalize` VS `MultithreadInfo::set_thread_limit()`, etc.

Reply:
`dealii::MultithreadInfo::set_thread_limit` sets the number of threads to the minimum its the argument and the environment variable. `TBB` is asked if the minimum is `dealii::numbers::invalid_unsigned_int`.
See [here](https://www.dealii.org/current/doxygen/deal.II/multithread__info_8cc_source.html#l00108). That function is called before `main` is executed.
 `dealii::Utilities::MPI::MPI_InitFinalize` just calls this function using its third argument (`dealii::numbers::invalid_unsigned_int` by default).

Consequently, the call here sets the number of threads to the one stored in the environment variable.